content/tutorials/reactor-firehose-kinesis.textile

```diff
@@ -0,0 +1,602 @@
+
+The code for this tutorial is written in *Node.js*, which you can "download here":https://nodejs.org/.
+
+h2. Step 1: Create your Kinesis data stream
+
+Execute the following @aws@ command in the terminal to create a Kinesis data stream called @sensor-data-stream@, with a single shard:
+
+```[sh]
+aws kinesis create-stream --stream-name sensor-data-stream --shard-count 1
+```
+
+Then, run the following @describe-stream@ command to view details of the stream and confirm that it has been created successfully:
+
+```[sh]
+aws kinesis describe-stream --stream-name sensor-data-stream
+```
+
+Make a note of the @StreamARN@ key's value. You will need it in the next and later steps:
+
+```[json]
+{
+    "StreamDescription": {
+        "Shards": [
+            {
+                "ShardId": "shardId-000000000000",
+                "HashKeyRange": {
+                    "StartingHashKey": "0",
+                    "EndingHashKey": "340282366920938463463374607431768211455"
+                },
+                "SequenceNumberRange": {
+                    "StartingSequenceNumber": "49623362581436102050070983950697146036412772107449008130"
+                }
+            }
+        ],
+   ==>  "StreamARN": "arn:aws:kinesis:eu-west-2:318417720898:stream/sensor-data-stream",
+        "StreamName": "sensor-data-stream",
+        "StreamStatus": "ACTIVE",
+        "RetentionPeriodHours": 24,
+        "EnhancedMonitoring": [
+            {
+                "ShardLevelMetrics": []
+            }
+        ],
+        "EncryptionType": "NONE",
+        "KeyId": null,
+        "StreamCreationTimestamp": "2021-10-27T11:21:31+01:00"
+    }
+}
+```
+
+h2. Step 2: Create an AWS role for Reactor Firehose
+
+So that Ably can write messages to your Kinesis data stream, you need to grant appropriate permissions. The safest way to do this, which does not require you to provide your AWS credentials, is to create an assumable role that the Ably platform can use.
+
+Our "AWS Authentication guide":/general/aws-authentication/ explains how to do this. Follow the steps in the guide to "create an AWS Kinesis policy":/general/aws-authentication/#create-kinesis-policy called @AblyKinesisPolicy@ and "attach it to a role":/general/aws-authentication/#create-role called @AblyKinesisRole@.
+
+Make a note of the @AblyKinesisRole@ ARN. You need this to configure your Firehose integration rule in the next step.
+
+h2. Step 3: Create the Reactor Firehose integration rule
+
+In this step you want to ensure that messages published by Ably to a specific channel (@raw-sensor-data@) are delivered to the Kinesis data stream called @sensor-data-stream@ that you created in Step 1.
+
+# Log into the "Ably dashboard":https://ably.com/login and click the Create New App button.
```

Comment:
I'd suggest bolding the UI interactions in this step, such as `Create New App`

Reply:
Good idea.

Reply:
Done